encodings/fastlanes/src/bitpacking/compress.rs

```diff
@@ -53,6 +77,9 @@ pub fn bitpack_primitive<T: NativePType + BitPacking>(array: &[T], bit_width: us
     let num_chunks = (array.len() + 1023) / 1024;
     let num_full_chunks = array.len() / 1024;
     let packed_len = 128 * bit_width / size_of::<T>();
+    // packed_len says how many values of size T we're going to include.
+    // 1024 * bit_width / 8 == the number of bytes we're going to get.
+    // then we divide by the size of T to get the number of elements.
 
     // Allocate a result byte array.
     let mut output = Vec::with_capacity(num_chunks * packed_len);
```

Comment:
for alignment: should just allocate this as a Buffer of the correct size (with Arrow), then assemble Vec::from_raw_parts (wrapped in ManuallyDrop), then have this function return the Buffer

Reply:
arrow buffers can `into_vec`